testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedRegistrationTest.java

```diff
@@ -0,0 +1,54 @@
+package org.keycloak.testsuite.actions;
+
+import jakarta.mail.MessagingException;
+import org.jboss.arquillian.graphene.page.Page;
+import org.junit.Before;
+import org.junit.Test;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.testsuite.AbstractTestRealmKeycloakTest;
+import org.keycloak.testsuite.Assert;
+import org.keycloak.testsuite.admin.ApiUtil;
+import org.keycloak.testsuite.pages.AppPage;
+import org.keycloak.testsuite.pages.RegisterPage;
+
+import java.io.IOException;
+
+public class AppInitiatedRegistrationTest extends AbstractTestRealmKeycloakTest {
+
+    @Page
+    protected AppPage appPage;
+
+    @Page
+    protected RegisterPage registerPage;
+
+    @Override
+    public void configureTestRealm(RealmRepresentation testRealm) {
+    }
+
+    @Before
+    public void before() {
+        ApiUtil.removeUserByUsername(testRealm(), "test-user@localhost");
+    }
+
+    @Test
+    public void ensureLocaleParameterIsPropagatedDuringAppInitiatedRegistration() throws IOException, MessagingException {
+
+        String appInitiatedRegisterUrl = oauth.getLoginFormUrl();
+        appInitiatedRegisterUrl = appInitiatedRegisterUrl.replace("openid-connect/auth", "openid-connect/registrations");
+
+        // add the kc_locale parameter
+        appInitiatedRegisterUrl += "&kc_locale=en";
+
+        driver.navigate().to(appInitiatedRegisterUrl);
+
+        registerPage.assertCurrent();
+        registerPage.register("first", "last", "test-user@localhost", "test-user", "test","test");
+
+        appPage.assertCurrent();
+
+        UserRepresentation user = testRealm().users().searchByEmail("test-user@localhost", true).get(0);
+        // ensure that the locale was set on the user
+        Assert.assertEquals("en", user.getAttributes().get("locale").get(0));
```

Comment:
Is English not the default locale assigned to a user? Perhaps using a more foreign locale would be a good way to test.

Reply:
the default should be the current language selected in login form,

Reply:
The local user attribute does not exist if the locale is not propagated properly.

I initially tested this with a german locale "de" however, then `appPage.assertCurrent();` breaks as it tries to match the english translation with a german title.

Hence I kept using "en".

Reply:
I applied the other suggestions to the PR.